serde/src/java/org/apache/hadoop/hive/serde2/JsonSerDe.java

```diff
@@ -139,7 +139,6 @@ private void initialize(final Configuration conf, final Properties tbl,
     this.jsonReader = new HiveJsonReader(this.soi, tsParser);
     this.jsonWriter = new HiveJsonWriter(this.binaryEncoding, getColumnNames());
 
-    this.jsonReader.setBinaryEncoding(binaryEncoding);
```

Comment:
can we drop the `binaryEncoding` var as well?

Reply:
@deniskuzZ Thanks for the review again!

Before [HIVE-21240](https://issues.apache.org/jira/browse/HIVE-21240), RAWSTRING was the default `binaryEncoding,` but it caused serialization issues. While most of these were addressed with precondition checks, switching to BASE64 binaryEncoding provides better handling for cases involving, say, special characters.

Moreover, if we decide to use RAWSTRING again in the future, the change can be made easily by updating a single variable, ensuring flexibility.

Reply:
sorry, my bad, I haven't noticed that we pass encoding to the writer as well